src/context/AuthProvider/reducer.ts

```diff
@@ -0,0 +1,16 @@
+import { createReducer } from '~utils/reducer';
+import { Action } from '~constants/interfaces/store';
+
+import { actions } from './actions';
+import { Auth, SetUserAction } from './interfaces';
+
+export const authState: Auth = {
+  currentUser: undefined
```

Comment:
I am not against this, but `undefined` has a very specific meaning in JS and setting it manually it's not quite semantic 👀I would use null instead.